flink-metrics/flink-metrics-core/src/main/java/org/apache/flink/metrics/groups/UnregisteredMetricsGroup.java

```diff
@@ -107,7 +107,7 @@ public MetricGroup addGroup(String key, String value) {
 
 	@Override
 	public Map<String, String> getAllVariables() {
-		return Collections.emptyMap();
+		return new HashMap<>();
```

Comment:
is this change necessary?

Reply:
I think it was before (tests were failing because I was using `...getAllVariables().put("shard_id")...`). I will revert and test.